src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,462 @@
+
+            var result = new DesignTimeInputsDelta
+            {
+                AddedItems = addedDesignTimeInputs.ToImmutable(),
+                RemovedItems = removedDesignTimeInputs.ToImmutable(),
+                AddedSharedItems = addedDesignTimeSharedInputs.ToImmutable(),
+                RootNamespace = rootNamespace,
+                OutputPath = outputPath,
+                // if this is the first time processing (previousOutput = null) then we will be "adding" all inputs
+                // but we don't want to immediately kick off a compile of all files. We'll let the events fire and
+                // the file timestamp will determine if we compile when someone asks for the TempPEs
+                ShouldCompile = (previousOutput != null),
+
+                DataSourceVersions = input.DataSourceVersions
+            };
+
+            return Task.FromResult(result);
+
+            bool PreprocessAddItem(string item)
+            {
+                ProjectItemInstance projItem = project.GetItemsByItemTypeAndEvaluatedInclude(Compile.SchemaName, item).FirstOrDefault();
+                System.Diagnostics.Debug.Assert(projItem != null, "Couldn't find the project item for Compile with an evaluated include of " + item);
+
+                bool added = false;
+                bool link = StringComparers.PropertyValues.Equals(projItem.GetMetadataValue(Compile.LinkProperty), bool.TrueString);
+                if (!link)
+                {
+                    bool designTime = StringComparers.PropertyValues.Equals(projItem.GetMetadataValue(Compile.DesignTimeProperty), bool.TrueString);
+                    bool designTimeShared = StringComparers.PropertyValues.Equals(projItem.GetMetadataValue(Compile.DesignTimeSharedInputProperty), bool.TrueString);
+
+                    if (designTime)
+                    {
+                        addedDesignTimeInputs.Add(item);
+                        added = true;
+                    }
+
+                    // Legacy allows files to be DesignTime and DesignTimeShared
+                    if (designTimeShared)
+                    {
+                        addedDesignTimeSharedInputs.Add(item);
+                        added = true;
+                    }
+                }
+                return added;
+            }
+        }
+    }
+
+    internal class DesignTimeInputsItem
+    {
+        public ImmutableHashSet<string> Inputs { get; set; } = ImmutableHashSet.Create(StringComparers.Paths);
+        public ImmutableHashSet<string> SharedInputs { get; set; } = ImmutableHashSet.Create(StringComparers.Paths);
+        public string OutputPath { get; internal set; }
+        public string RootNamespace { get; internal set; }
+    }
+
+    internal class DesignTimeInputsDelta
+    {
+        public bool ShouldCompile { get; set; } = true;
+        public ImmutableArray<string> AddedSharedItems { get; set; } = ImmutableArray.Create<string>();
```

Comment:
While the same thing, we usually use "Empty" for things like this so that someone doesn't think "Oh does Create return a new value?".